### PR TITLE
fix(components): [table] cell tooltip display error

### LINKED ISSUE #16673: [Component] [table-column] Problems occur when el-table-column uses show-overflow-tooltip at a specific resolution
Bug Type: **`Component`**

## Environment

- Vue Version: `3.3.10`
- Element Plus Version: `2.6.3`
- Browser / OS: `Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36 Edg/123.0.0.0`
- Build Tool: `Vite`

### Related Component

- `el-table-column`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGVsLXRhYmxlIDpkYXRhPVwidGFibGVEYXRhXCIgc3R5bGU9XCJ3aWR0aDogMTAwJVwiIGhlaWdodD1cIjI1MFwiPlxuICAgIDxlbC10YWJsZS1jb2x1bW4gc2hvdy1vdmVyZmxvdy10b29sdGlwIGZpeGVkIHByb3A9XCJkYXRlXCIgbGFiZWw9XCJEYXRlXCIgd2lkdGg9XCIxNTBcIiAvPlxuICAgIDxlbC10YWJsZS1jb2x1bW4gc2hvdy1vdmVyZmxvdy10b29sdGlwIHByb3A9XCJuYW1lXCIgbGFiZWw9XCJOYW1lXCIgd2lkdGg9XCIxMjBcIiAvPlxuICAgIDxlbC10YWJsZS1jb2x1bW4gc2hvdy1vdmVyZmxvdy10b29sdGlwIHByb3A9XCJzdGF0ZVwiIGxhYmVsPVwiU3RhdGVcIiB3aWR0aD1cIjEyMFwiIC8+XG4gICAgPGVsLXRhYmxlLWNvbHVtbiBzaG93LW92ZXJmbG93LXRvb2x0aXAgcHJvcD1cImNpdHlcIiBsYWJlbD1cIkNpdHlcIiB3aWR0aD1cIjMyMFwiIC8+XG4gICAgPGVsLXRhYmxlLWNvbHVtbiBzaG93LW92ZXJmbG93LXRvb2x0aXAgcHJvcD1cImFkZHJlc3NcIiBsYWJlbD1cIkFkZHJlc3NcIiB3aWR0aD1cIjYwMFwiIC8+XG4gICAgPGVsLXRhYmxlLWNvbHVtbiBzaG93LW92ZXJmbG93LXRvb2x0aXAgcHJvcD1cInppcFwiIGxhYmVsPVwiWmlwXCIgd2lkdGg9XCIxMjBcIiAvPlxuICA8L2VsLXRhYmxlPlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmNvbnN0IHRhYmxlRGF0YSA9IFtcbiAge1xuICAgIGRhdGU6ICcyMDE2LTA1LTAzJyxcbiAgICBuYW1lOiAnVG9tJyxcbiAgICBzdGF0ZTogJ0NhbGlmb3JuaWEnLFxuICAgIGNpdHk6ICdMb3MgQW5nZWxlcycsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gICAgemlwOiAnQ0EgOTAwMzYnLFxuICB9LFxuICB7XG4gICAgZGF0ZTogJzIwMTYtMDUtMDInLFxuICAgIG5hbWU6ICdUb20nLFxuICAgIHN0YXRlOiAnQ2FsaWZvcm5pYScsXG4gICAgY2l0eTogJ0xvcyBBbmdlbGVzJyxcbiAgICBhZGRyZXNzOiAnTm8uIDE4OSwgR3JvdmUgU3QsIExvcyBBbmdlbGVzJyxcbiAgICB6aXA6ICdDQSA5MDAzNicsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wNCcsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgc3RhdGU6ICdDYWxpZm9ybmlhJyxcbiAgICBjaXR5OiAnTG9zIEFuZ2VsZXMnLFxuICAgIGFkZHJlc3M6ICdOby4gMTg5LCBHcm92ZSBTdCwgTG9zIEFuZ2VsZXMnLFxuICAgIHppcDogJ0NBIDkwMDM2JyxcbiAgfSxcbiAge1xuICAgIGRhdGU6ICcyMDE2LTA1LTAxJyxcbiAgICBuYW1lOiAnVG9tJyxcbiAgICBzdGF0ZTogJ0NhbGlmb3JuaWEnLFxuICAgIGNpdHk6ICdMb3MgQW5nZWxlcycsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gICAgemlwOiAnQ0EgOTAwMzYnLFxuICB9LFxuICB7XG4gICAgZGF0ZTogJzIwMTYtMDUtMDgnLFxuICAgIG5hbWU6ICdUb20nLFxuICAgIHN0YXRlOiAnQ2FsaWZvcm5pYScsXG4gICAgY2l0eTogJ0xvcyBBbmdlbGVzJyxcbiAgICBhZGRyZXNzOiAnTm8uIDE4OSwgR3JvdmUgU3QsIExvcyBBbmdlbGVzJyxcbiAgICB6aXA6ICdDQSA5MDAzNicsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wNicsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgc3RhdGU6ICdDYWxpZm9ybmlhJyxcbiAgICBjaXR5OiAnTG9zIEFuZ2VsZXMnLFxuICAgIGFkZHJlc3M6ICdOby4gMTg5LCBHcm92ZSBTdCwgTG9zIEFuZ2VsZXMnLFxuICAgIHppcDogJ0NBIDkwMDM2JyxcbiAgfSxcbiAge1xuICAgIGRhdGU6ICcyMDE2LTA1LTA3JyxcbiAgICBuYW1lOiAnVG9tJyxcbiAgICBzdGF0ZTogJ0NhbGlmb3JuaWEnLFxuICAgIGNpdHk6ICdMb3MgQW5nZWxlcycsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gICAgemlwOiAnQ0EgOTAwMzYnLFxuICB9LFxuXVxuPC9zY3JpcHQ+XG4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiBcIkVTTmV4dFwiLFxuICAgIFwianN4XCI6IFwicHJlc2VydmVcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcInR5cGVzXCI6IFtcImVsZW1lbnQtcGx1cy9nbG9iYWwuZC50c1wiXSxcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWUsXG4gICAgXCJhbGxvd0pzXCI6IHRydWUsXG4gICAgXCJjaGVja0pzXCI6IHRydWVcbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59XG4iLCJfbyI6e319)

### Steps to reproduce

The screen resolution is 2560 * 1600.
El-table-column added show-overflow-tooltip.
Enlarge (110% or other) or shrink (90% or other) the browser.

### What is Expected?

Zoom in or out the resolution. Content that is not exceeded by el-table should not display tooltip.

### What is actually happening?

All the contents of the table do not overflow, but there will be tooltip

### Additional comments

_(empty)_

### LINKED ISSUE #16856: [Component] [table-column] 浏览器缩放110%，table文字长度不到列宽一半，鼠标悬浮却显示了tooltip
Bug Type: **`Component`**

## Environment

- Vue Version: `3.4.27`
- Element Plus Version: `2.7.3`
- Browser / OS: `Chrome 124.0.0.0 Windows 10`
- Build Tool: `Vite`

### Related Component

- `el-table-column`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGVsLXRhYmxlXG4gICAgcmVmPVwibXVsdGlwbGVUYWJsZVJlZlwiXG4gICAgOmRhdGE9XCJ0YWJsZURhdGFcIlxuICAgIHN0eWxlPVwid2lkdGg6IDEwMCVcIlxuICAgIEBzZWxlY3Rpb24tY2hhbmdlPVwiaGFuZGxlU2VsZWN0aW9uQ2hhbmdlXCJcbiAgPlxuICAgIDxlbC10YWJsZS1jb2x1bW4gdHlwZT1cInNlbGVjdGlvblwiIHdpZHRoPVwiNTVcIiAvPlxuICAgIDxlbC10YWJsZS1jb2x1bW4gbGFiZWw9XCJEYXRlXCIgd2lkdGg9XCIxMjBcIj5cbiAgICAgIDx0ZW1wbGF0ZSAjZGVmYXVsdD1cInNjb3BlXCI+e3sgc2NvcGUucm93LmRhdGUgfX08L3RlbXBsYXRlPlxuICAgIDwvZWwtdGFibGUtY29sdW1uPlxuICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcGVydHk9XCJuYW1lXCIgbGFiZWw9XCJOYW1lXCIgd2lkdGg9XCIxMjBcIiAvPlxuICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcGVydHk9XCJhZGRyZXNzXCIgbGFiZWw9XCJBZGRyZXNzXCIgc2hvdy1vdmVyZmxvdy10b29sdGlwIC8+XG4gIDwvZWwtdGFibGU+XG4gIDxkaXYgc3R5bGU9XCJtYXJnaW4tdG9wOiAyMHB4XCI+XG4gICAgPGVsLWJ1dHRvbiBAY2xpY2s9XCJ0b2dnbGVTZWxlY3Rpb24oW3RhYmxlRGF0YVsxXSwgdGFibGVEYXRhWzJdXSlcIj5cbiAgICAgIFRvZ2dsZSBzZWxlY3Rpb24gc3RhdHVzIG9mIHNlY29uZCBhbmQgdGhpcmQgcm93c1xuICAgIDwvZWwtYnV0dG9uPlxuICAgIDxlbC1idXR0b24gQGNsaWNrPVwidG9nZ2xlU2VsZWN0aW9uKClcIj5DbGVhciBzZWxlY3Rpb248L2VsLWJ1dHRvbj5cbiAgPC9kaXY+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c1wiIHNldHVwPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgRWxUYWJsZSB9IGZyb20gJ2VsZW1lbnQtcGx1cydcblxuaW50ZXJmYWNlIFVzZXIge1xuICBkYXRlOiBzdHJpbmdcbiAgbmFtZTogc3RyaW5nXG4gIGFkZHJlc3M6IHN0cmluZ1xufVxuXG5jb25zdCBtdWx0aXBsZVRhYmxlUmVmID0gcmVmPEluc3RhbmNlVHlwZTx0eXBlb2YgRWxUYWJsZT4+KClcbmNvbnN0IG11bHRpcGxlU2VsZWN0aW9uID0gcmVmPFVzZXJbXT4oW10pXG5jb25zdCB0b2dnbGVTZWxlY3Rpb24gPSAocm93cz86IFVzZXJbXSkgPT4ge1xuICBpZiAocm93cykge1xuICAgIHJvd3MuZm9yRWFjaCgocm93KSA9PiB7XG4gICAgICAvLyBUT0RPOiBpbXByb3ZlbWVudCB0eXBpbmcgd2hlbiByZWZhY3RvciB0YWJsZVxuICAgICAgLy8gZXNsaW50LWRpc2FibGUtbmV4dC1saW5lIEB0eXBlc2NyaXB0LWVzbGludC9iYW4tdHMtY29tbWVudFxuICAgICAgLy8gQHRzLWV4cGVjdC1lcnJvclxuICAgICAgbXVsdGlwbGVUYWJsZVJlZi52YWx1ZSEudG9nZ2xlUm93U2VsZWN0aW9uKHJvdywgdW5kZWZpbmVkKVxuICAgIH0pXG4gIH0gZWxzZSB7XG4gICAgbXVsdGlwbGVUYWJsZVJlZi52YWx1ZSEuY2xlYXJTZWxlY3Rpb24oKVxuICB9XG59XG5jb25zdCBoYW5kbGVTZWxlY3Rpb25DaGFuZ2UgPSAodmFsOiBVc2VyW10pID0+IHtcbiAgbXVsdGlwbGVTZWxlY3Rpb24udmFsdWUgPSB2YWxcbn1cblxuY29uc3QgdGFibGVEYXRhOiBVc2VyW10gPSBbXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wMycsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wMicsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wNCcsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wMScsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wOCcsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wNicsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wNycsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gIH0sXG5dXG48L3NjcmlwdD5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIl9vIjp7fX0=)

### Steps to reproduce

直接用文档里多选的示例，因为这是文档里第一次介绍show-overflow-tooltip属性
image

### What is Expected?

cell内容不超出不显示tooltip

### What is actually happening?

总是显示tooltip

### Additional comments

浏览器放大110%，这个示例代码就能复现

### PR DESCRIPTION
**Pull request by btea**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

fix #16673 
fix #16856

### CODE REVIEW

File: packages/components/table/src/table-body/events-helper.ts [hunk 1]
```diff
@@ -8,6 +8,10 @@ import type { TableColumnCtx } from '../table-column/defaults'
 import type { TableBodyProps } from './defaults'
 import type { TableOverflowTooltipOptions } from '../util'
 
+function compareValueWithError(a: number, b: number, error = 0.01) {
```
Comment (plainheart) [inline packages/components/table/src/table-body/events-helper.ts:11]: I would suggest using a clearer name like `isGreaterThan` and a more common name for the `error` parameter like `epsilon`.

File: packages/components/table/src/table-body/events-helper.ts [hunk 2]
```diff
@@ -8,6 +8,10 @@ import type { TableColumnCtx } from '../table-column/defaults'
...
+  return a > b && a - b > error
```
Comment (plainheart) [inline packages/components/table/src/table-body/events-helper.ts:12]: It seems `a - b > error` is enough. What is the reason why you added two checks?

Comment (btea) [reply packages/components/table/src/table-body/events-helper.ts:12]: Thank you for your review, I seem to have made unnecessary judgments. 😅

File: packages/components/table/src/table-body/events-helper.ts (hunk 1 above)
Comment (btea) [reply packages/components/table/src/table-body/events-helper.ts]: Good suggestion, I've made changes.

**Review by plainheart — APPROVED** [review]

**Review by kooriookami — APPROVED** [review]

**Comment by rikako1021** [comment]
Sorry but is this fix really working correctly?
I also had these problems (like #16673 #16856) and since the latest version 2.7.4 was released, I upgraded and built again
 but I still have same problem.
I still get tooltip on hover in all columns with `show-overflow-tooltip` in el-table.

Vue 3.4.13
OS: Windows
Browser: Chrome
Build Tool: Vite

following two have already been done.
`rm -rf node_modules`
`rm package-lock.json`

**Comment by btea** [comment]
This problem is essentially caused by precision loss. This PR tolerates a certain degree of precision loss and does solve the related issue.
table

**Comment by rikako1021** [comment]
This is my code, I have the same problem on all columns of table.
```
<el-table :data="datafoo">
      <el-table-column prop="userId" label="column1" show-overflow-tooltip width="200px"/>
      <el-table-column prop="userName" label="column2" show-overflow-tooltip width="250px"/>
      <el-table-column prop="userEmail" label="column3" show-overflow-tooltip min-width="300px"/>
</el-table>
```


🖋 Note that Element's SCSS specifies the `font-size` of the table row in this way, and removing this specification eliminated the unnecessary ToolTip display.
Does this mean that not only the X-direction but also the Y-direction overflow is involved in determining whether a ToolTip is displayed or not?
If I don't want to change the `font-size`, what should I use to adjust it..?
.el-table__row {
  font-size: 16px;
}

**Comment by btea** [comment]
> 🖋 Note that Element's SCSS specifies the `font-size` of the table row in this way, and removing this specification eliminated the unnecessary ToolTip display. Does this mean that not only the X-direction but also the Y-direction overflow is involved in determining whether a ToolTip is displayed or not?

Yes

It looks like you changed the cell's font-size so that the content height exceeds the cell height?

**Comment by rikako1021** [comment]
Yes.
Strictly, if the `font-size` exceeds the height of the class name `'cell el-tooltip'`, will the tooltip appear? So to prevent the tooltip from appearing without changing `font-size`, should I adjust the height of this class?

**Comment by btea** [comment]
Yes.

**Comment by rikako1021** [comment]
I could fix problem by adjusting `cell` heights! Thank you.

**Comment by ZhangYC193** [comment]
vue3 +  @vite,element-plus@2.7.4+,chrom浏览器l-table-column缩放80%，也会在不占满区域的情况下出现气泡。 @rikako1021 @btea

**Pull request merged**